commit 218f4cefea087eadcd8ddf47d719beda1cb6370b
Author: Sage Weil <sweil@redhat.com>
Date:   Fri May 28 22:19:27 2021 -0400

    mgr: make mgr commands compat with pre-quincy mon
    
    If the mon is not yet quincy, do not share the quincy-ism of
    'positional=false'.
    
    Signed-off-by: Sage Weil <sage@newdream.net>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 1ba6eaea4f6..14d95ab5158 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -12,6 +12,7 @@
  */
 
 #include <Python.h>
+#include <boost/algorithm/string/replace.hpp>
 
 #include "common/errno.h"
 #include "common/signal.h"
@@ -258,6 +259,12 @@ void MgrStandby::send_beacon()
       std::vector<MonCommand> commands = mgr_commands;
       std::vector<MonCommand> py_commands = py_module_registry.get_commands();
       commands.insert(commands.end(), py_commands.begin(), py_commands.end());
+      if (monc.monmap.min_mon_release < ceph_release_t::quincy) {
+	dout(10) << " stripping out positional=false quincy-ism" << dendl;
+	for (auto& i : commands) {
+	  boost::replace_all(i.cmdstring, ",positional=false", "");
+	}
+      }
       m->set_command_descs(commands);
       dout(4) << "going active, including " << m->get_command_descs().size()
               << " commands in beacon" << dendl;
